pkg/k8s/secrets_manager.go

```diff
@@ -0,0 +1,133 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/util/workqueue"
+	"sync"
+
+	"github.com/go-logr/logr"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+)
+
+// SecretsManager manages the secret resources needed by the controller
+type SecretsManager interface {
+	// MonitorSecrets manages the individual watches for the given secrets
+	MonitorSecrets(ingressGroupID string, secrets []types.NamespacedName)
+}
+
+func NewSecretsManager(clientSet kubernetes.Interface, eventHandler handler.EventHandler, logger logr.Logger) *defaultSecretsManager {
+	return &defaultSecretsManager{
+		mutex:        sync.Mutex{},
+		secretMap:    make(map[types.NamespacedName]*secretItem),
+		eventHandler: eventHandler,
+		clientSet:    clientSet,
+		logger:       logger,
+	}
+}
+
+var _ SecretsManager = &defaultSecretsManager{}
+
+type defaultSecretsManager struct {
+	mutex        sync.Mutex
+	secretMap    map[types.NamespacedName]*secretItem
+	eventHandler handler.EventHandler
+	clientSet    kubernetes.Interface
+	queue        workqueue.RateLimitingInterface
```

Comment:
seems this queue is not initialized.

Reply:
this is not used by the secrets eventhandler

Reply:
used channel instead